src/TimeModeling/judiVector.jl

```diff
@@ -713,19 +713,25 @@ iterate(S::judiVector, state::Integer=1) = state > S.nsrc ? nothing : (S.data[st
 
 function cumsum(x::judiVector;dims=1)
     dims == 1 || dims == 2 || throw(judiVectorException("Dimension $(dims) is out of range for a 2D array"))
-    y = deepcopy(x)
+    y = (dims == 1 ? x.geometry.dt[1] : (x.geometry.xloc[1][2]-x.geometry.xloc[1][1])) * x
```

Comment:
In addition to what felix said on slack, this would only work for uniformly sampled receiver along the x coordinate only in 2D which is very restrictive.